boa/src/object/internal_methods.rs

```diff
@@ -424,14 +422,16 @@ impl GcObject {
                     return None;
                 }
 
-                let char = if let Some(utf16_val) = string.encode_utf16().nth(pos) {
-                    Value::from(from_u32(utf16_val as u32).unwrap())
+                let result_str = if let Some(utf16_val) = string.encode_utf16().nth(pos) {
+                    Value::from(char::from_u32(utf16_val as u32).unwrap())
                 } else {
                     return None;
                 };
 
-                let desc =
-                    PropertyDescriptor::from(DataDescriptor::new(char, Attribute::ENUMERABLE));
+                let desc = PropertyDescriptor::from(DataDescriptor::new(
+                    result_str,
+                    Attribute::ENUMERABLE,
```

Comment:
Here, I would add the rest of the `false` flags, even though the result will be the same. It's more explicit and makes things more understandable:
```suggestion
                    Attribute::READONLY | Attribute::ENUMERABLE | Attribute::PERMANENT,
```